### PR TITLE
Update hetznercloud driver

### PR DESCRIPTION
**Pull request by 6543**
- log caller if level is debug or trace
- bump hetznercloud client lib to v2
- don't panic on wrong labels settings
- select image arch based server type

### CODE REVIEW

**Comment by 6543** [comment]
as one can see the deploy of arm works now -> https://ci.woodpecker-ci.org/repos/8975/pipeline/60/7

but the image is not ready jet :D

**Comment by xoxys** [comment]
We should enable renovate bot or similar across the WP org. These outdated dependencies all over the place are somewhat annoying.

File: engine/stringmaps.go [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
 func SliceToMap(list []string, del string) map[string]string {
 	m := make(map[string]string)
 	for _, e := range list {
 		parts := strings.Split(e, del)
-		m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+		if len(parts) == 2 {
+			m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+		} else {
+			log.Debug().Msgf("could not split '%s' into key value pair with '=' delimiter", e)
```
Comment (anbraten) [inline engine/stringmaps.go:16]: I would throw an error instead to notify the user

File: providers/hetznercloud/driver.go [hunk 2]
```diff
@@ -144,12 +144,17 @@ func (d *Driver) DeployAgent(ctx context.Context, agent *woodpecker.Agent) error
...
+	serverType, _, err := d.client.ServerType.GetByName(ctx, d.ServerType)
+	if err != nil {
+		return fmt.Errorf("%s: ServerTypeGetByName: %w", d.Name, err)
 	}
 
-	image, _, err := d.client.Image.GetByName(ctx, d.Image)
+	image, _, err := d.client.Image.GetByNameAndArchitecture(ctx, d.Image, serverType.Architecture)
 	if err != nil {
-		return fmt.Errorf("%s: %w", d.Image, err)
+		return fmt.Errorf("%s: GetByName: %w", d.Image, err)
```
Comment (anbraten) [inline providers/hetznercloud/driver.go:157]: get by name and architecture

**Comment by xoxys** [comment]
> We should enable **renovate bot** or similar 

Can recommend it, and it's open source.

**Comment by 6543** [comment]
Can agree on renovate ... in this case it was a majour bump witch renovate does not do anyway ...

**Comment by xoxys** [comment]
It does, https://github.com/thegeeklab/drone-github-comment/pull/254

**Review by xoxys — APPROVED** [review]

**Review by pat-s — APPROVED** [review]

**Pull request merged**